### ISSUE TITLE
Scrobble not working

### ISSUE THREAD

**Issue by MaskedTrickster** [kepelet/flo#30]
I have it setup and works fine when listening straight through navidromes web ui but nothing shows when listening through flo

**Comment by MaskedTrickster** [comment]
i use listenbrainz i should clarify

**Comment by faultables** [comment]
Navidrome's scrobble implementation is client-side (it means the client needs to call /scrobble explicitly), and flo currently has not supported it yet. But we're getting closer into it!

**Comment by MaskedTrickster** [comment]
okay that makes sense i thought it did as its on the website but thats fine there are other apps that do it tho like amperfy and substreamer i just dont like the ui of thoes apps and they seems to preform not well

**Comment by faultables** [comment]
yeah it's understandable. i use listenbrainz too and still scrobbling it on the navidrome web ui

**Comment by sansfontieres** [comment]
delaying scrobbles if the device is offline OR if the server is not reachable would be great too :>

**Comment by faultables** [comment]
> delaying scrobbles if the device is offline OR if the server is not reachable would be great too :>

yeah that would be great especially for offline streaming

**Comment by jostyee** [comment]
The latest TestFlight doesn't support scrobbling downloaded songs.

**Comment by faultables** [comment]
> The latest TestFlight doesn't support scrobbling downloaded songs.

scrobbling needs to call https://your-navidrome-server/rest/scrobble endpoint to work. however, you can still scrobble the downloaded songs as long as you're logged in

**Comment by jostyee** [comment]
@faultables sorry I meant playing in offline mode, it won't resume scrobbling when back online.

**Comment by faultables** [comment]
> [@faultables](https://github.com/faultables) sorry I meant playing in offline mode, it won't resume scrobbling when back online.

@jostyee ah i know what you mean. currently the scrobbling only happens once you start the track (to update the "now playing" status in the scrobble service) and when track is >= 50% complete (submitting to the scrobble service). if the server goes back online in the middle of playback, i believe flo will submit the scrobble

would you share your situation specifically?

**Comment by jostyee** [comment]
> > [@faultables](https://github.com/faultables) sorry I meant playing in offline mode, it won't resume scrobbling when back online.
> 
> [@jostyee](https://github.com/jostyee) ah i know what you mean. currently the scrobbling only happens once you start the track (to update the "now playing" status in the scrobble service) and when track is >= 50% complete (submitting to the scrobble service). if the server goes back online in the middle of playback, i believe flo will submit the scrobble
> 
> would you share your situation specifically?

Like flying without internet, or I get a bad signal for a while on subway. Clients such as Symfonium for Android cache offline play history and resume submitting them when back online, so I'd not lose it.

**Comment by faultables** [comment]
> > > [@faultables](https://github.com/faultables) sorry I meant playing in offline mode, it won't resume scrobbling when back online.
> > 
> > 
> > [@jostyee](https://github.com/jostyee) ah i know what you mean. currently the scrobbling only happens once you start the track (to update the "now playing" status in the scrobble service) and when track is >= 50% complete (submitting to the scrobble service). if the server goes back online in the middle of playback, i believe flo will submit the scrobble
> > would you share your situation specifically?
> 
> Like flying without internet, or I get a bad signal for a while on subway. Clients such as Symfonium for Android cache offline play history and resume submitting them when back online, so I'd not lose it.

gotcha, thanks! i will plan that feature in 1.7
